internal/controller/postgrescluster/instance.go

```diff
@@ -559,6 +566,12 @@ func (r *Reconciler) reconcileInstanceSets(
 		return err
 	}
 
+	// Cleanup Instance Set resources that are no longer needed
+	err = r.deleteOrphanedPodDisruptionBudgets(ctx, cluster)
```

Comment:
:pick: "Orphaned" in a Kube context makes me think resources that don't have an owner, which doesn't really apply here IIUC ... maybe can just be something like `cleanupPodDisruptionBudgets()`?